src/coreclr/src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.CoreCLR.cs

```diff
@@ -27,7 +27,7 @@ private static IArraySortHelper<T> CreateArraySortHelper()
 
             if (typeof(IComparable<T>).IsAssignableFrom(typeof(T)))
             {
-                defaultArraySortHelper = (IArraySortHelper<T>)RuntimeTypeHandle.Allocate(typeof(GenericArraySortHelper<string>).TypeHandle.Instantiate(new Type[] { typeof(T) }));
+                defaultArraySortHelper = (IArraySortHelper<T>)typeof(GenericArraySortHelper<string>).TypeHandle.Instantiate(new Type[] { typeof(T) }).CreateInstanceDefaultCtor(publicOnly: false, skipCheckThis: true, fillCache: false, wrapExceptions: false)!;
```

Comment:
Wonder if this can be modified to use `RuntimeHandles.CreateInstanceForAnotherGenericParameter`? Should be fairly straightforward to convert that method to a QCALL and to allow it to take an arbitrary number of type handles as the generic argument.